doschema/__init__.py

```diff
@@ -22,7 +22,82 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""JSON Schema utility functions and commands."""
+"""JSON Schema utility functions and commands.
+
+Compatibility Validation
+-------------------------
+
+It validates compatibility between different JSON schemas versions.
+
+A schema is backward compatible if the fields' type remain the same in all
+JSON schemas declaring it and JSON schemas are type consistent within
+themselves too.
+
+>>> import json
+>>> import os
+>>> from io import open
+>>>
+>>> import chardet
+>>> import click
+>>>
+>>> import doschema.validation
+>>>
+>>> schemas = ['jsonschema_f1.json', 'jsonschema_f2.json']
+>>>
+>>> schema_validator = doschema.validation.JSONSchemaValidator()
+>>> for schema in schemas:
+...     path = os.path.join(os.getcwd(), 'tests', 'data_files', schema)
+...     with open(path, 'rb') as infile:
+...         byte_file = infile.read()
+...         encoding = chardet.detect(byte_file)['encoding']
+...         if encoding in ['UTF-16BE', 'UTF-16LE']:
+...             encoding = 'UTF-16'
+...         elif encoding in ['UTF-32BE', 'UTF-32LE']:
+...             encoding = 'UTF-32'
+...         string_file = byte_file.decode(encoding)
+...         schema = json.loads(string_file)
+...         schema_validator.validate(schema, path)
+
+By default the index of "array" "items" are ignored. Thus all the values of
+an array should have the same type in order to be compatible.
+This behavior can be disabled by setting "ignore_index = False" in the
+validator's constructor.
+
+>>> import json
+>>> import os
+>>> from io import open
+>>>
+>>> import chardet
+>>> import click
+>>>
+>>> import doschema.validation
+>>>
+>>> schemas = ['jsonschema_f1.json', 'jsonschema_f2.json']
+>>>
+>>> schema_validator = doschema.validation.JSONSchemaValidator(
+...     ignore_index = False
+... )
+>>> for schema in schemas:
+...     path = os.path.join(os.getcwd(), 'tests', 'data_files', schema)
+...     with open(path, 'rb') as infile:
+...         byte_file = infile.read()
+...         encoding = chardet.detect(byte_file)['encoding']
+...         if encoding in ['UTF-16BE', 'UTF-16LE']:
+...             encoding = 'UTF-16'
+...         elif encoding in ['UTF-32BE', 'UTF-32LE']:
+...             encoding = 'UTF-32'
```

Comment:
I see that the encoding detection code is duplicated here and in `cli.py`. Could you factorize this code in a `detect_encoding` function in a separate `utils.py` file?